OndaEDFSchemas.jl/src/OndaEDFSchemas.jl

```diff
@@ -1,94 +1,135 @@
-                kind::Union{Missing,AbstractString}
-                channel::Union{Missing,AbstractString}
-                sample_unit::Union{Missing,AbstractString}
-                sample_resolution_in_unit::Union{Missing,LPCM_SAMPLE_TYPE_UNION}
-                sample_offset_in_unit::Union{Missing,LPCM_SAMPLE_TYPE_UNION}
-                sample_type::Union{Missing,AbstractString}
-                sample_rate::Union{Missing,LPCM_SAMPLE_TYPE_UNION}
-                # errors, use `nothing` to indicate no error
-                error::Union{Nothing,String})
+@schema "ondaedf.plan" Plan
+
+@version PlanV1 begin
+    # EDF.SignalHeader fields
+    label::String
+    transducer_type::String
+    physical_dimension::String
+    physical_minimum::Float32
+    physical_maximum::Float32
+    digital_minimum::Float32
+    digital_maximum::Float32
+    prefilter::String
+    samples_per_record::Int16
+    # EDF.FileHeader field
+    seconds_per_record::Float64
+    # Onda.SamplesInfo fields (channels -> channel), may be missing
+    kind::Union{Missing,AbstractString} = lift(String, kind)
+    channel::Union{Missing,AbstractString} = lift(String, channel)
+    sample_unit::Union{Missing,AbstractString} = lift(String, sample_unit)
+    sample_resolution_in_unit::Union{Missing,LPCM_SAMPLE_TYPE_UNION} = lift(convert_number_to_lpcm_sample_type, sample_resolution_in_unit)
+    sample_offset_in_unit::Union{Missing,LPCM_SAMPLE_TYPE_UNION} = lift(convert_number_to_lpcm_sample_type, sample_offset_in_unit)
+    sample_type::Union{Missing,AbstractString} = lift(onda_sample_type_from_julia_type, sample_type)
+    sample_rate::Union{Missing,LPCM_SAMPLE_TYPE_UNION} = lift(convert_number_to_lpcm_sample_type, sample_rate)
+    # errors, use `nothing` to indicate no error
+    error::Union{Nothing,String} = coalesce(error, nothing)
+end
+
+@version PlanV2 begin
+    # EDF.SignalHeader fields
+    label::String
+    transducer_type::String
+    physical_dimension::String
+    physical_minimum::Float32
+    physical_maximum::Float32
+    digital_minimum::Float32
+    digital_maximum::Float32
+    prefilter::String
+    samples_per_record::Int16
+    # EDF.FileHeader field
+    seconds_per_record::Float64
+    # Onda.SamplesInfo fields (channels -> channel), may be missing
+    sensor_type::Union{Missing,AbstractString} = lift(String, sensor_type)
+    channel::Union{Missing,AbstractString} = lift(String, channel)
+    sample_unit::Union{Missing,AbstractString} = lift(String, sample_unit)
+    sample_resolution_in_unit::Union{Missing,LPCM_SAMPLE_TYPE_UNION} = lift(convert_number_to_lpcm_sample_type, sample_resolution_in_unit)
+    sample_offset_in_unit::Union{Missing,LPCM_SAMPLE_TYPE_UNION} = lift(convert_number_to_lpcm_sample_type, sample_offset_in_unit)
+    sample_type::Union{Missing,AbstractString} = lift(onda_sample_type_from_julia_type, sample_type)
+    sample_rate::Union{Missing,LPCM_SAMPLE_TYPE_UNION} = lift(convert_number_to_lpcm_sample_type, sample_rate)
```

Comment:
I think these fields have different types now in SignalV2: 

https://github.com/beacon-biosignals/Onda.jl/blob/1c69482315700342c706c1c4d4800231c0bc93e9/src/signals.jl#L80-L83